### PR TITLE
Add caching to prevent redundant creation of textures on GPU

### PR DESCRIPTION
**Pull request by errissa**
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/4317)

### CODE REVIEW

File: cpp/open3d/visualization/rendering/filament/FilamentResourceManager.cpp [hunk 1]
```diff
@@ -441,10 +456,16 @@ TextureHandle FilamentResourceManager::CreateTexture(const char* path,
 TextureHandle FilamentResourceManager::CreateTexture(
         const std::shared_ptr<geometry::Image>& img, bool srgb) {
     TextureHandle handle;
-    if (img->HasData()) {
+    auto hash = fnv1a_block_hash(img->data_.data(), img->data_.size());
```
Comment (ssheorey) [inline cpp/open3d/visualization/rendering/filament/FilamentResourceManager.cpp:459]: Are uint8 / float / double images supported? The hash function assumes int data here.

**Review by ssheorey — COMMENTED** [review]
Curious if just `img->data_.data()` or `hash(data(), width, height)` works as well. They should be much faster to calculate. For (data(), width, height), `std::unordered_map<tuple, image>` should work directly.

File: cpp/open3d/visualization/rendering/filament/FilamentResourceManager.cpp (hunk 1 above)
Comment (errissa) [reply cpp/open3d/visualization/rendering/filament/FilamentResourceManager.cpp:459]: The hash function takes a void* and treats the memory block as a block of octets so it should work with any image format.

**Comment by errissa** [comment]
> Curious if just `img->data_.data()` or `hash(data(), width, height)` works as well. They should be much faster to calculate. For (data(), width, height), `std::unordered_map<tuple, image>` should work directly.

Perhaps in the future but with the current necessary translations between legacy and new materials new memory sometimes gets allocated so `img->data_.data()` is not always the same for the same image.

**Review by ssheorey — APPROVED** [review]
This greatly reduces memory consumption in the TB plugin with the demo scene. GPU memory is around 720MB on the first load, increases by about 30MB on every reload. CPU memory usage is about 1.7GB (all of tensorboard) on first load, and increases by about 50MB per reload.

Solves the Out of Memory error after 2 reloads on my laptop.

**Pull request merged**